lib/cylc/config.py

```diff
@@ -274,6 +274,21 @@ def __init__(self, suite, fpath, template_vars=[],
         else:
             self.initial_point.standardise()
 
+        # Validate initial cycle point against any constraints
+        if self.cfg['scheduling']['initial cycle point constraints']:
```

Comment:
The logical test is there so that the following code won't run if the list is empty i.e. there are no constraints to compare against. I can't come up with a way that wouldn't require comparing against this variable at some point or testing that the expanded version of it was non-empty.